tests/sentry/logging/test_logging_exceptions.py

```diff
@@ -0,0 +1,38 @@
+import logging
+from unittest.mock import Mock
+
+import pytest
+import structlog
+from django.conf import settings
+from django.test import override_settings
+
+from sentry.logging.handlers import JSONRenderer, StructLogHandler
+from sentry.testutils.cases import TestCase
+
+
+class TestExceotionsDuringLogging(TestCase):
+    """
+    Test that exceptions during logging are handled gracefully in Production,
+    while still being surfaced in non-Production environments.
+    """
+
+    def setUp(self):
+        super().setUp()
+        structlog.configure(processors=[JSONRenderer()])
+        self.logger = logging.getLogger(__name__)
+        self.logger.addHandler(StructLogHandler())
+        self.logger.propagate = False
+        self.foo = Mock()
+
+    @override_settings(IS_PROD=False)
+    def test_non_prod(self):
+        logging.raiseExceptions = (
+            not settings.IS_PROD
+        )  # TODO: this is a hack, figure out how to test this properly e2e
```

Comment:
this permanently mutates a global variable and never undoes it -- you want to use `with mock.patch.object(...)` to restore the value after a context